### DISCUSSION TITLE
With img2img, is it possible to use my own custom depth map?

### DISCUSSION THREAD

**Discussion by bruibbruib** [Mikubill/sd-webui-controlnet#195] [Q&A] [answered]
With img2img, is it possible to use my own, better quality depth map? I load my depth map into ControlNet, but can't understand how to utilise it. It always just smushes it, or just generates one on the fly.
Thanks for any help!!

**Comment by ljleb** [comment]
To input a custom preprocessed detectmap image of any type, you have to select `none` for the preprocessor and use the appropriate controlnet model (the depth model in your case). By selecting `none` for the preprocessor, you tell the extension to use the detectmap input as is.

**Reply by bruibbruib** [reply]
That was my interpretation of how to use it, and I did try it that way, but the output was not right...
I will try again now and verify. There have been updates since.
Thank you for your reply.

**Reply by bruibbruib** [reply]
I could be missing something, but still not working. Here are my settings:
![hfgdhtrdht](https://user-images.githubusercontent.com/82312962/219881892-c22dbfba-3419-417f-bf13-86d088fac4f4.png)
and the output (image and depth):
![yhtjufkuh](https://user-images.githubusercontent.com/82312962/219881958-bdb8c909-fdde-438c-bb22-d1298eb14710.png)
If I remove the CN input image, and set PRE to depth, I get a more expected output, but not using my depthmap:
![jhgjhgfjhg](https://user-images.githubusercontent.com/82312962/219881989-371a5ec0-2d44-472a-bd82-fdf7a2691d1f.png)

Any Hints?!

**Comment by bruibbruib** [comment, marked as answer]
Answered my own question.
My custom detectmap was in .png, greyscale, 16bit format.
Changed it to 8bit png, or jpeg and it works as expected.

**Reply by bruibbruib** [reply]
More is revealed:
While the above is true, 8 bit maps look ruff man.
Deeper investigation reveals I needed to save as sRGB and no transparency to keep the .png at 16bit, and smooth.

**Reply by AugmentedRealityCat** [reply]
You can still use part of the 16 bit data if you load it in an image editing app and play with the levels / curves to extract the part you want to work with before saving it in 8bit. For the cat picture example, you could crash all the darker depth values at the back to near pure black, and do the same in white for whats nearer the camera, and keep all the 256 values range you get in 8 bit for the cat itself. That being said, the depth map you used is already pretty much like that, and that technique would not help much, but it's something to know.

Also, there are some known [bugs with the Adobe code managing 16 bit PNG](https://github.com/AnonymousCervine/depth-image-io-for-SDWebui/issues/2#event-8513039602)s that actually interprets them as 15 bits, and this can cause problems in some specific cases. If you are using photoshop to edit 16 bit greyscale PNG, then you could have problems. The workaround I found was to use Krita instead for those specific cases.

**Reply by bruibbruib** [reply]
Top Tips! I had no idea about photoshop issues.
Thanks for the info.
